src/graph_gui.cpp

```diff
@@ -557,6 +564,16 @@ struct BaseGraphWindow : Window {
 
 				x += x_sep;
 			}
+		};
+
+		/* Draw non-highlighted datasets. */
+		for (const DataSet &dataset : this->data) {
+			if (dataset.exclude_bit != this->highlight_data && dataset.range_bit != this->highlight_range) draw_dataset(dataset);
+		}
+
+		/* Draw highlighted datasets. */
+		for (const DataSet &dataset : this->data) {
+			if (dataset.exclude_bit == this->highlight_data || dataset.range_bit == this->highlight_range) draw_dataset(dataset);
 		}
```

Comment:
Why iterate twice instead of drawing highlighted and not in the same loop?

Reply:
So that the highlighted dataset is drawn last, i.e. on top of the reset.